### PR TITLE
Added UT for callback controller.

### PR DESCRIPTION
**Pull request by yashgit891**
### - Link added should be a valid URL and accessible throughout the org.
   ### - If the branch name contains hotfix / revert by default the BVT workflow check will pass.

| Test Case Document URL                        |
|-----------------------------------------------|
| Please paste test case document link here.... |

### CODE REVIEW

File: Test/Unit/CallbackControllerTest.php [hunk 1]
```diff
@@ -0,0 +1,382 @@
...
+                                                                                                $this->logger,
+                                                                                                $this->orderRepository,
+                                                                                                $this->transaction,
+                                                                                                $this->invoiceService,
+                                                                                                $this->orderSender,
+                                                                                                $this->invoiceSender,
+                                                                                                $this->catalogSession,
+                                                                                                $this->order])->makePartial()->shouldAllowMockingProtectedMethods();
+        $this->callback->shouldReceive('getRequest')->andReturn($this->requestInterface);
+
```
Comment (garimaprachi) [inline Test/Unit/CallbackControllerTest.php]: Remove line gaps before function close and add one after function close

File: Test/Unit/CallbackControllerTest.php [hunk 2]
```diff
@@ -0,0 +1,382 @@
...
+        $this->requestInterface->shouldReceive('getParams')->andReturn(['order_id' => '000012',
+                                                                        'razorpay_payment_id' => 'Test Payment id']);
+        $this->callback->objectManagement = \Mockery::mock(
+            \Magento\Framework\App\ObjectManager::class
+        );
+        $this->callback->captureCommand = $this->captureCommand;
+        $this->captureCommand->shouldReceive('execute');
+        $this->callback->objectManagement->shouldReceive('get')->with('Magento\Sales\Model\Order')->andReturn($this->orderCollection);
+        $this->callback->objectManagement->shouldReceive('get')->with('Magento\Quote\Model\Quote')->andReturn($this->quoteModel);
+        //var_dump($this->callback->objectManagement);
```
Comment (garimaprachi) [inline Test/Unit/CallbackControllerTest.php]: Remove commented line

File: Test/Unit/CallbackControllerTest.php [hunk 3]
```diff
@@ -0,0 +1,382 @@
...
+        $this->requestInterface->shouldReceive('getParams')->andReturn(['order_id' => '000012',
+                                                                        'razorpay_payment_id' => 'Test Payment id']);
+        $this->callback->objectManagement = \Mockery::mock(
+            \Magento\Framework\App\ObjectManager::class
+        );
+        $this->callback->authorizeCommand = $this->authorizeCommand;
+        $this->authorizeCommand->shouldReceive('execute');
+        $this->callback->objectManagement->shouldReceive('get')->with('Magento\Sales\Model\Order')->andReturn($this->orderCollection);
+        $this->callback->objectManagement->shouldReceive('get')->with('Magento\Quote\Model\Quote')->andReturn($this->quoteModel);
+        //var_dump($this->callback->objectManagement);
```
Comment (garimaprachi) [inline Test/Unit/CallbackControllerTest.php]: Remove commented line

File: Test/Unit/CallbackControllerTest.php [hunk 4]
```diff
@@ -0,0 +1,382 @@
...
+        $this->assertSame($this->response,$this->callback->execute());
```
Comment (garimaprachi) [inline Test/Unit/CallbackControllerTest.php]: Add space after comma

File: Test/Unit/CallbackControllerTest.php [hunk 5]
```diff
@@ -0,0 +1,382 @@
...
+
+        $this->requestInterface->shouldReceive('getParams')->andReturn(['order_id' => '']);
+        $this->callback->objectManagement = \Mockery::mock(
+            \Magento\Framework\App\ObjectManager::class
+        );
+        $this->callback->captureCommand = $this->captureCommand;
+        $this->captureCommand->shouldReceive('execute');
+        $this->callback->objectManagement->shouldReceive('get')->with('Magento\Sales\Model\Order')->andReturn($this->orderCollection);
+        $this->callback->objectManagement->shouldReceive('get')->with('Magento\Quote\Model\Quote')->andReturn($this->quoteModel);
+        //var_dump($this->callback->objectManagement);
```
Comment (garimaprachi) [inline Test/Unit/CallbackControllerTest.php]: Remove commented line

File: Test/Unit/CallbackControllerTest.php [hunk 6]
```diff
@@ -0,0 +1,382 @@
...
+        $this->requestInterface->shouldReceive('getParams')->andReturn(['order_id' => '000012',
+                                                                        'razorpay_payment_id' => 'Test Payment id']);
+        $this->callback->objectManagement = \Mockery::mock(
+            \Magento\Framework\App\ObjectManager::class
+        );
+        $this->callback->captureCommand = $this->captureCommand;
+        $this->captureCommand->shouldReceive('execute');
+        $this->callback->objectManagement->shouldReceive('get')->with('Magento\Sales\Model\Order')->andReturn($this->orderCollection);
+        $this->callback->objectManagement->shouldReceive('get')->with('Magento\Quote\Model\Quote')->andReturn($this->quoteModel);
+        //var_dump($this->callback->objectManagement);
```
Comment (garimaprachi) [inline Test/Unit/CallbackControllerTest.php]: Remove commented line

File: Test/Unit/CallbackControllerTest.php [hunk 7]
```diff
@@ -0,0 +1,382 @@
...
+        $this->callback->execute();
+    }
```
Comment (garimaprachi) [inline Test/Unit/CallbackControllerTest.php:292]: Add line after function end

File: Test/Unit/CallbackControllerTest.php [hunk 8]
```diff
@@ -0,0 +1,382 @@
...
+    public function testErrorFromValidateSignature()
+    {
+
```
Comment (garimaprachi) [inline Test/Unit/CallbackControllerTest.php]: Remove extra line

File: Test/Unit/CallbackControllerTest.php [hunk 9]
```diff
@@ -0,0 +1,382 @@
...
+        $this->requestInterface->shouldReceive('getParams')->andReturn(['order_id' => '000012',
+                                                                        'razorpay_payment_id' => 'Test Payment id']);
+        $this->callback->objectManagement = \Mockery::mock(
+            \Magento\Framework\App\ObjectManager::class
+        );
+        $this->callback->captureCommand = $this->captureCommand;
+        $this->captureCommand->shouldReceive('execute');
+        $this->callback->objectManagement->shouldReceive('get')->with('Magento\Sales\Model\Order')->andReturn($this->orderCollection);
+        $this->callback->objectManagement->shouldReceive('get')->with('Magento\Quote\Model\Quote')->andReturn($this->quoteModel);
+        //var_dump($this->callback->objectManagement);
```
Comment (garimaprachi) [inline Test/Unit/CallbackControllerTest.php]: Remove commented line

File: Test/Unit/CallbackControllerTest.php [hunk 10]
```diff
@@ -0,0 +1,382 @@
...
+        $this->callback->execute();
+    }
```
Comment (garimaprachi) [inline Test/Unit/CallbackControllerTest.php]: Add line after function end

File: Test/Unit/CallbackControllerTest.php [hunk 11]
```diff
@@ -0,0 +1,382 @@
...
+        $this->callback->captureCommand = $this->captureCommand;
+        $this->captureCommand->shouldReceive('execute');
+        $this->callback->objectManagement->shouldReceive('get')->with('Magento\Sales\Model\Order')->andReturn($this->orderCollection);
+        $this->callback->objectManagement->shouldReceive('get')->with('Magento\Quote\Model\Quote')->andReturn($this->quoteModel);
+        //var_dump($this->callback->objectManagement);
+        $this->callback->execute();
+    } 
+    public function testExecuteFailure()
+    {
+
```
Comment (garimaprachi) [inline Test/Unit/CallbackControllerTest.php]: Remove extra blank line

File: Test/Unit/CallbackControllerTest.php [hunk 12]
```diff
@@ -0,0 +1,382 @@
...
+        $this->requestInterface->shouldReceive('getParams')->andReturn(['order_id' => '000012']);
+        $this->callback->objectManagement = \Mockery::mock(
+            \Magento\Framework\App\ObjectManager::class
+        );
+        $this->callback->captureCommand = $this->captureCommand;
+        $this->captureCommand->shouldReceive('execute');
+        $this->callback->objectManagement->shouldReceive('get')->with('Magento\Sales\Model\Order')->andReturn($this->orderCollection);
+        $this->callback->objectManagement->shouldReceive('get')->with('Magento\Quote\Model\Quote')->andReturn($this->quoteModel);
+        //var_dump($this->callback->objectManagement);
```
Comment (garimaprachi) [inline Test/Unit/CallbackControllerTest.php]: Remove commented line

File: Test/Unit/CallbackControllerTest.php [hunk 13]
```diff
@@ -0,0 +1,382 @@
...
+        $this->assertSame($this->response, $this->callback->execute());
+    }
```
Comment (garimaprachi) [inline Test/Unit/CallbackControllerTest.php:334]: Add line after function end

File: Test/Unit/CallbackControllerTest.php [hunk 14]
```diff
@@ -0,0 +1,382 @@
...
+        $this->requestInterface->shouldReceive('getParams')->andReturn(['order_id' => '000012',
+                                                                        'razorpay_payment_id' => 'Test Payment id']);
+        $this->callback->objectManagement = \Mockery::mock(
+            \Magento\Framework\App\ObjectManager::class
+        );
+        $this->callback->captureCommand = $this->captureCommand;
+        $this->captureCommand->shouldReceive('execute');
+        $this->callback->objectManagement->shouldReceive('get')->with('Magento\Sales\Model\Order')->andReturn($this->orderCollection);
+        $this->callback->objectManagement->shouldReceive('get')->with('Magento\Quote\Model\Quote')->andReturn($this->quoteModel);
+        //var_dump($this->callback->objectManagement);
```
Comment (garimaprachi) [inline Test/Unit/CallbackControllerTest.php]: Remove commented line

File: Test/Unit/CallbackControllerTest.php [hunk 15]
```diff
@@ -0,0 +1,382 @@
...
+        $this->assertSame($this->response, $this->callback->execute());
+    }
```
Comment (garimaprachi) [inline Test/Unit/CallbackControllerTest.php:357]: Add blank line after function end

File: Test/Unit/CallbackControllerTest.php [hunk 16]
```diff
@@ -0,0 +1,382 @@
...
+        $this->requestInterface->shouldReceive('getParams')->andReturn(['order_id' => '000012',
+                                                                        'razorpay_payment_id' => 'Test Payment id']);
+        $this->callback->objectManagement = \Mockery::mock(
+            \Magento\Framework\App\ObjectManager::class
+        );
+        $this->callback->captureCommand = $this->captureCommand;
+        $this->captureCommand->shouldReceive('execute');
+        $this->callback->objectManagement->shouldReceive('get')->with('Magento\Sales\Model\Order')->andReturn($this->orderCollection);
+        $this->callback->objectManagement->shouldReceive('get')->with('Magento\Quote\Model\Quote')->andReturn($this->quoteModel);
+        //var_dump($this->callback->objectManagement);
```
Comment (garimaprachi) [inline Test/Unit/CallbackControllerTest.php]: Remove commented line

File: Test/Unit/CallbackControllerTest.php [hunk 17]
```diff
@@ -0,0 +1,382 @@
...
+        $this->assertSame($this->response, $this->callback->execute());
+    }
```
Comment (garimaprachi) [inline Test/Unit/CallbackControllerTest.php:379]: Add line after function end

File: Test/Unit/CallbackControllerTest.php [hunk 18]
```diff
@@ -0,0 +1,382 @@
...
+    public function testvalidateSignatureFailure()
+    {
+        $request = [
+            'error' => 'Test error'
+        ];
+
+        $this->expectException("Exception");
+        $this->expectExceptionMessage("Payment Failed or error from gateway");
+        $this->callback->validateSignature($request);
+    }
```
Comment (garimaprachi) [inline Test/Unit/CallbackControllerTest.php:390]: Add line after function end

File: Test/Unit/CallbackControllerTest.php [hunk 19]
```diff
@@ -0,0 +1,382 @@
...
+                    case 'utility':
+                        return $this->utilityApi;
+                }
+            }
+        );
+        $this->callback->razorpayOrderID = 'Test razorpay order id';
+        $this->callback->rzp = $this->api;
+        $this->callback->validateSignature($request);
+    }
+}
```
Comment (garimaprachi) [inline Test/Unit/CallbackControllerTest.php]: Add line after end of file

File: Test/Unit/CallbackControllerTest.php (hunk 4 above)
Comment (garimaprachi) [reply Test/Unit/CallbackControllerTest.php]: @enishant

File: Test/Unit/CallbackControllerTest.php (hunk 19 above)
Comment (garimaprachi) [reply Test/Unit/CallbackControllerTest.php]: @enishant

**Review by garimaprachi — COMMENTED** [review]
LGTM

File: Test/Unit/CallbackControllerTest.php [hunk 20]
```diff
@@ -0,0 +1,399 @@
...
+        $this->requestInterface->shouldReceive('getParams')->andReturn(['order_id' => '000012',
+                                                                        'razorpay_payment_id' => 'Test Payment id']);
+        $this->callback->objectManagement = \Mockery::mock(
+            \Magento\Framework\App\ObjectManager::class
+        );
+        $this->callback->captureCommand = $this->captureCommand;
+        $this->captureCommand->shouldReceive('execute');
+        $this->callback->objectManagement->shouldReceive('get')->with('Magento\Sales\Model\Order')->andReturn($this->orderCollection);
+        $this->callback->objectManagement->shouldReceive('get')->with('Magento\Quote\Model\Quote')->andReturn($this->quoteModel);
+        $this->assertSame($this->response, $this->callback->execute());
```
Comment (garimaprachi) [inline Test/Unit/CallbackControllerTest.php:223]: Add line before assertion

**Review by garimaprachi — COMMENTED** [review]
LGTM

File: Test/Unit/CallbackControllerTest.php [hunk 21]
```diff
@@ -0,0 +1,422 @@
+<?php
+declare(strict_types=1);
+
+use PHPUnit\Framework\TestCase;
+use \Magento\Framework\Exception\MailException;
+use Magento\Framework\Phrase;
+
+class CallbackControllerTest extends TestCase {
```
Comment (abdulwahidsharief) [inline Test/Unit/CallbackControllerTest.php]: @yashgit891 flower bracket should start with new line

**Review by abdulwahidsharief — APPROVED** [review]
LGTM

**Pull request merged**